tests/foreman/cli/test_syncplan.py

```diff
@@ -273,3 +319,271 @@ def test_verify_bugzilla_1261122(self):
         new_sync_plan = self._make_sync_plan()
         result = SyncPlan.info({'id': new_sync_plan['id']})
         self.assertIsNotNone(result.get('enabled'))
+
+    # This Bugzilla bug is private. It is impossible to fetch info about it.
+    @stubbed('Unstab when BZ1279539 is fixed')
+    @tier1
+    def test_verify_bugzilla_1279539(self):
+        """Verify product won't get synced immediately after adding association
+        with a sync plan which has already been started
+
+        @Feature: Sync Plan
+
+        @Assert: Repository was not synchronized
+
+        @BZ: 1279539
+        """
+        sync_plan = self._make_sync_plan({
+            'enabled': 'true',
+            'organization-id': self.org['id'],
+            'sync-date': datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
+        })
+        product = make_product({'organization-id': self.org['id']})
+        repo = make_repository({'product-id': product['id']})
+        Product.set_sync_plan({
+            'id': product['id'],
+            'sync-plan-id': sync_plan['id'],
+        })
+        with self.assertRaises(AssertionError):
+            self.validate_repo_content(
```

Comment:
Since this is a negative test and the repo is not going to be synced how much ever time we wait here, I would vote for max_attempts=3 here.